spec/integration/api/conversations_controller_spec.rb

```diff
@@ -0,0 +1,122 @@
+require "spec_helper"
+
+
+describe Api::V0::ConversationsController do
+  let(:auth) { FactoryGirl.create(:auth_with_read_and_write) }
+  let!(:access_token) { auth.create_access_token.to_s }
+
+  before do
+    auth.user.share_with bob.person, auth.user.aspects[0]
+    auth.user.share_with alice.person, auth.user.aspects[0]
+    alice.share_with auth.user.person, alice.aspects[0]
+
+    @conversation = {
+      author_id: auth.user.id,
+      subject: "new conversation",
+      text: "first message",
+      messages: [{
+        text: "first message"
+      }],
+      recipients: [alice.person.id],
+      access_token: access_token
+    }
+  end
+
+  describe "#create" do
+    context "with valid data" do
+      it "creates the conversation" do
+        post(
+          api_v0_conversations_path(), 
+          @conversation
+        )
+        expect(response.status).to eq 201
+        response_body = JSON.parse(response.body)["conversation"]
+        expect(response_body["messages"][0]["id"]).to_not be_nil
+        expect(response_body["id"]).to_not be_nil
+        expect(response_body["participants"].length).to eq 2
+      end
+    end
+
+    context "without valid data" do
+      it "fails at creating the conversation" do
+        post api_v0_conversations_path(), { :access_token => access_token }
```

Comment:
- Do not use parentheses for method calls with no arguments.
- Redundant curly braces around a hash parameter.
- Space inside { detected.
- Use the new Ruby 1.9 hash syntax.
- Space inside } detected.